File: Leetcode/Python/1437.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        dist = float("inf")
        prev = -1
        for i in range(len(nums)):
            if nums[i] == 1:
                if dist < k:
                    return False
                prev = i
                dist = 0
            else:
                dist += 1
        return True

File: Leetcode/c/1437.c
bool kLengthApart(int* nums, int numsSize, int k) { 
    int prev = -1;
    int dist = INT_MAX;
    for (int i = 0; i < numsSize; ++i) {
        if (nums[i] == 1) {
            if (dist < k) {
                return false;
            }
            dist = 0;
            prev = i;
        } else {
            if (dist != INT_MAX) {
                dist++;
            };
        }
    }
    return true;
}
